### PR TITLE
#538 Backport analysis fixes to 2.1.1.

### PR DESCRIPTION
**Pull request by zooba**
#538

### CODE REVIEW

File: Python/Product/Django/TemplateParsing/TemplateArtifactCollection.cs [hunk 1]
```diff
@@ -29,11 +29,14 @@ public TemplateArtifactCollection()
             LeftSeparator = RightSeparator = "";
         }
 
-        protected override ISensitiveFragmentSeparatorsInfo SeparatorInfo {
-            get {
-                return this;
-            }
-        }
+        // HACK: Does not build properly on Dev12
```
Comment (crwilcox) [inline Python/Product/Django/TemplateParsing/TemplateArtifactCollection.cs]: is this interesting for this case?  Also, there is another bit of dead code (analysis dictionary) we might want to remove in both 2.2 and 2.1

**Comment by crwilcox** [comment]
:+1:  Provided all the tests are run and show okay.  I scoured the code and nothing is too alarming.  It really shouldn't be since you just ported some changes to a branch.

File: Python/Product/Django/TemplateParsing/TemplateArtifactCollection.cs (hunk 1 above)
Comment (zooba) [inline Python/Product/Django/TemplateParsing/TemplateArtifactCollection.cs]: Not really, except I was working on this in VS 2013 and so not being able to build was really annoying. This code is ifdef'd out for VS 2010 and VS 2012, so it's irrelevant.

AnalysisDictionary is not dead code - it's used for keeping track of variables. If anything, it's probably our biggest bottleneck (which is why changing it to a mostly thread-safe dictionary from a regular dictionary is such a good thing).

**Comment by int19h** [comment]
:+1:

**Pull request merged**